From 6f9e80c2ac09247cfff69e6c43d06bc559978a3b Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 7 Jan 2015 16:04:51 -0500
Subject: [iommu] vt-d: Only remove domain when device is removed

Message-id: <20150107160451.9952.76929.stgit@amt.stowe>
Patchwork-id: 102519
O-Subject: [RHEL7.1 PATCH 2/3] iommu/vt-d: Only remove domain when device is removed
Bugzilla: 1109829
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1109829
Upstream Status: 1196c2fb0407683c2df92d3d09f9144d42830894

commit 1196c2fb0407683c2df92d3d09f9144d42830894
Author: Joerg Roedel <jroedel@suse.de>
Date:   Tue Sep 30 13:02:03 2014 +0200

    iommu/vt-d: Only remove domain when device is removed

    This makes sure any RMRR mappings stay in place when the
    driver is unbound from the device.

    Signed-off-by: Joerg Roedel <jroedel@suse.de>
    Tested-by: Jerry Hoemann <jerry.hoemann@hp.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 8d020f2..b2f9aa7 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -3860,8 +3860,7 @@ static int device_notifier(struct notifier_block *nb,
  if (iommu_dummy(dev))
   return 0;
 
- if (action != BUS_NOTIFY_UNBOUND_DRIVER &&
-     action != BUS_NOTIFY_DEL_DEVICE)
+ if (action != BUS_NOTIFY_REMOVED_DEVICE)
   return 0;
 
  domain = find_domain(dev);
-- 
1.7.1